instrumentation/rocketmq-client-4.8/testing/src/main/java/base/IntegrationTestBase.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package base;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+import java.util.concurrent.atomic.AtomicInteger;
+import org.apache.rocketmq.broker.BrokerController;
+import org.apache.rocketmq.common.BrokerConfig;
+import org.apache.rocketmq.common.namesrv.NamesrvConfig;
+import org.apache.rocketmq.logging.InternalLogger;
+import org.apache.rocketmq.logging.InternalLoggerFactory;
+import org.apache.rocketmq.namesrv.NamesrvController;
+import org.apache.rocketmq.remoting.netty.NettyClientConfig;
+import org.apache.rocketmq.remoting.netty.NettyServerConfig;
+import org.apache.rocketmq.store.config.MessageStoreConfig;
+import org.apache.rocketmq.test.util.MQAdmin;
+import org.junit.Assert;
+
+public class IntegrationTestBase {
```

Comment:
The whole rocketmq setup & start procedure seems to be pretty complicated. Do you think you can use testcontainers instead to just run a docker image? That's what we usually try to do in our tests when we need to use some external dependency/service

Reply:
Hi @mateuszrzeszutek ,
The official of testcontainers didn't provide the module of RocketMQ. It is not easy to customize testcontainers anywhere to run RocketMQ  docker image for dependency. For example, in this condition, brokerIP1 usually is suggested to be configurated to physical IP. It is easier to use current rocketmq-test library ways.